include/listformatter.h

```diff
@@ -17,18 +18,21 @@ class ListFormatter {
 	void set_line(const unsigned int itempos, const StflRichText& text);
 	void clear()
 	{
+		std::lock_guard<std::mutex> guard(mutex);
 		lines.clear();
 	}
-	std::string format_list() const;
-	unsigned int get_lines_count() const
+	std::string format_list();
+	unsigned int get_lines_count()
 	{
+		std::lock_guard<std::mutex> guard(mutex);
 		return lines.size();
 	}
 
 private:
 	std::vector<StflRichText> lines;
 	RegexManager* rxman;
 	std::string location;
+	std::mutex mutex;
```

Comment:
You could mark the mutex [mutable](https://en.cppreference.com/w/cpp/language/cv#mutable_specifier) to restore the `const` for the methods.

```suggestion
	mutable std::mutex mutex;
```